### DISCUSSION TITLE
(ffa) Correlation

### DISCUSSION THREAD

**Discussion by V-ktor** [redeclipse/base#1176] [Maps and Content Showcase]
This is a medium sized map with an open layout and large room. It's made for ffa mode. ctf is also supported but very unbalanced (not recommended).

**r173 RC2:**
- some geometry changes
- removed teleporters and yellow teleporter room
- added more buildings outside the map
- new, flat glass material
- added actor entities (roller!)
- changes to lighting at mapeffects 1 and 3
- texture fixes
- adjusted pushers for plasma/rifle users

**r146 RC1:**
- added scenery geometry
- added geometry for pick up locations
- added sounds
- disable few particles and mapmodels at mapeffects 1 and 2

**v3:**
- new weapon placement
- some texture changes
- some other tweaks

Map file: 
[correlation.zip](https://github.com/redeclipse/world/files/3668037/correlation.zip)

Screenshots:
![20190929222421](https://user-images.githubusercontent.com/12966454/65839053-0e439900-e30a-11e9-8721-0ae7b437be8e.jpg)
![20190929222432](https://user-images.githubusercontent.com/12966454/65839055-0f74c600-e30a-11e9-8ab2-63df95b6c3ca.jpg)
![20190929222526](https://user-images.githubusercontent.com/12966454/65839056-10a5f300-e30a-11e9-9c27-b2954894abee.jpg)

**Comment by no-lex** [comment]
![20180626120410](https://user-images.githubusercontent.com/23445929/41933627-4c8f1626-7939-11e8-98b1-923674737377.png)
This opening is awkward to smoothly travel through; it's too high to single-jump through, and double jumping leads to hitting one's head on the top bar.

![20180626115713](https://user-images.githubusercontent.com/23445929/41933628-4d0c457e-7939-11e8-80f0-3ddb933e09d2.png)
The round grates here are too far out, should be an easy fix.

![20180626120320](https://user-images.githubusercontent.com/23445929/41933631-4dd12c18-7939-11e8-8939-acfe9bfa3599.png)
Over here, the ceiling bars have no glass, while the other side does; I found this somewhat confusing since the map is similar on both sides, and I was somewhat expecting glass here.

![20180626120019](https://user-images.githubusercontent.com/23445929/41934302-46da5e32-793b-11e8-97d4-9c3bba637f08.png)
I was a little confused about what these doors were until they teleported me. Use of particle effects or portal-type parallax is customary in other RE maps, and may help clarify what is going on here.

![20180626122314](https://user-images.githubusercontent.com/23445929/41934497-b988383c-793b-11e8-9996-8a60b498c2d4.png)
I'm not a huge fan of this jumppad; it feels like it's been stuck on the window rather than being a legitimate feature and is rather unintuitive.

![20180626122958](https://user-images.githubusercontent.com/23445929/41934886-c5129cbe-793c-11e8-9edd-f1934abaed25.png)
Floating, non-supported stairs made of handwavium. I tend to run a single center brace on stairs like these, but beware of inefficient geometry. If it's too much hassle, the current situation is probably adequate.


When I work on maps like this, I always try to make sure that similar features on different sides of the map are colored differently so that it is instantly possible to orient oneself on the map. There are several spots on this map where it is difficult to figure out where you are on the map; naturally, since you made the map, this will not be an issue for you, but for other people playing on the map, it could be confusing.

Otherwise, the map is aesthetically quite nice, and is not too far from release quality. The layout is also pretty good, though totally unsuited for CTF; the verticality is quite welcome.

**Comment by V-ktor** [comment]
Thank you for the feedback.

I must have forgotten that stairs. They have a slope of 1/3 so it's a bit annoying to think of suitable geometry.
I'll make sure to solve the other problems as well.

**Comment by cloudziq** [comment]
yeah, I also spotted these windows, glass texture is not consistent on these, most of the windows use another texture than on a few other panels. Also it's just a joke that teleport leads from one base to another? (grab a flag and you can score a point two seconds later)
Otherwise it's a good designed map.
and for round grates I can only suggest remaking these models with added custom collisions, or use almost invisible alpha geometry there

**Comment by V-ktor** [comment]
Here's a new version:
[correlation.zip](https://github.com/red-eclipse/world/files/2167962/correlation.zip)
![20180705212052](https://user-images.githubusercontent.com/12966454/42343844-ac1e8a3c-809a-11e8-85e0-8bec81b71191.jpg)

As for the glass texture, I used glass material where it is possible and a glass texture otherwise. It's not possible to make a texture looking exactly or very similar as glass material. I guess I could still try to make them more similar.
The teleporters are blocked with door models for ctf. It does not make sense to play something else than ffa on this map but I still added affinities for ctf, dac and bb.

**Comment by no-lex** [comment]
The release candidate is a pretty decent upgrade from your previous release, but there's a handful of little things i'd like to point out:

45 degree stairs and a square-aligned texture don't line up well:
![20190927130156](https://user-images.githubusercontent.com/23445929/65798898-b9482d00-e127-11e9-9efe-fc7d2e217861.png)

Not quite consistent extrusion texturing on the horizontal surfaces:
![20190927131134](https://user-images.githubusercontent.com/23445929/65798980-e1379080-e127-11e9-8691-b303074be55d.png)
![20190927131129](https://user-images.githubusercontent.com/23445929/65798981-e1d02700-e127-11e9-8699-ba4848984c39.png)

Grain runs the wrong way across the beam here (vertical surfaces oriented correctly but horizontal runs across):
![20190927130643](https://user-images.githubusercontent.com/23445929/65799038-0926f400-e128-11e9-921e-b8c9affb3bf7.png)

Some material volumes are of dubious efficiency (just looks a bit sloppy in edit mode):
![20190927130553](https://user-images.githubusercontent.com/23445929/65799093-222fa500-e128-11e9-9b35-47bea4e194b0.png)

You can see this grate from within the map:
![20190927130446](https://user-images.githubusercontent.com/23445929/65799392-ddf0d480-e128-11e9-84cf-7b8e4832aa3a.png)

I'm not sure about the "cracked" default glass on this map, perhaps instead use a flat texture (with refract disabled)?
![20190927130220](https://user-images.githubusercontent.com/23445929/65799398-e5b07900-e128-11e9-802d-10ac25f3c8dc.png)

The wall glass looks quite different than the material glass, which is a bit jarring (i'd use an more strongly envmapping texture; i use flat white at ~`.2 .2 .2 vcolour` to properly tint the outside)
![20190927130240](https://user-images.githubusercontent.com/23445929/65799462-0678ce80-e129-11e9-8ab0-a231ab04593f.png)

I'd add a few more buildings; the ones you added are a welcome addition:
![20190927132413](https://user-images.githubusercontent.com/23445929/65799800-d4b43780-e129-11e9-9a36-dd38ebd9f9c6.png)

Whole bunch of warning texture voffset issues:
![20190927132812](https://user-images.githubusercontent.com/23445929/65800130-8e130d00-e12a-11e9-9d12-d9a2749e7e20.png)
![20190927132806](https://user-images.githubusercontent.com/23445929/65800131-8e130d00-e12a-11e9-91b7-e2a4807f4986.png)
![20190927132802](https://user-images.githubusercontent.com/23445929/65800132-8eaba380-e12a-11e9-9e9b-dcfe960c532a.png)
![20190927132754](https://user-images.githubusercontent.com/23445929/65800133-8eaba380-e12a-11e9-8910-1de190d5d23d.png)
![20190927132749](https://user-images.githubusercontent.com/23445929/65800134-8eaba380-e12a-11e9-8004-11a1b291cc1f.png)
![20190927132744](https://user-images.githubusercontent.com/23445929/65800135-8eaba380-e12a-11e9-9c8d-fc04fe522aa6.png)

Still not sure if this is obviously a jumper; also doesn't work any more with current physics:
![20190927130538](https://user-images.githubusercontent.com/23445929/65800183-a8e58180-e12a-11e9-9311-67119b9cc81c.png)

The affinities still make no sense, so just remove them. This won't get merged as anything other than a FFA map (as you indicated yourself). I'm also not certain this map needs any teleporters considering its lack of extremities that might need them.
![20190927133258](https://user-images.githubusercontent.com/23445929/65800285-e8ac6900-e12a-11e9-96a6-579c58334d5e.png)

Some glass material slop here:
![20190927133421](https://user-images.githubusercontent.com/23445929/65800316-04177400-e12b-11e9-82cd-bfacccf55c21.png)

This jumper's not quite powerful enough for rifle+plasma loadouts:
![20190927134029](https://user-images.githubusercontent.com/23445929/65800995-d03d4e00-e12c-11e9-9ce8-da2b29f53700.png)


The map effects at max `mapeffects` is hardly a drop at all from minimum `mapeffects`, so you probably should be less conservative with your lighting at higher settings levels. I only noted a difference of about 10%. The minimum `mapeffects` performance appears to be mostly adequate, but you will never get a lot of dynamic lights on a map with no occlusion like this one and stay within the `mapeffects 0` performance requirements.

The lighting at `mapeffects 3` should have all point lights cast shadows (not be flagged static) if they physically can. This is a dynamically lit engine, and at max settings the dynamic lighting should come to the fore.

This is a pretty well textured and styled map nonetheless, and I only have some reservations about the complexity of the layout and the unnecessary concessions for team play, which this map is not remotely adequate at nonetheless.
